src/object/cli_obj.c

```diff
@@ -5140,6 +5141,11 @@ obj_csum_update(struct dc_object *obj, daos_obj_update_t *args, struct obj_auxi_
 	if (!obj_csum_dedup_candidate(&obj->cob_co->dc_props, args->iods, args->nr))
 		return 0;
 
+	if (obj_auxi->csum_retry) {
+		/* Release old checksum result and prepare for new calculation */
+		daos_csummer_free_ic(obj->cob_co->dc_csummer, &obj_auxi->rw_args.iod_csums);
+	}
+
 	return dc_obj_csum_update(obj->cob_co->dc_csummer, obj->cob_co->dc_props,
 				  obj->cob_md.omd_id, args->dkey, args->iods, args->sgls, args->nr,
 				  obj_auxi->reasb_req.orr_singv_los, &obj_auxi->rw_args.dkey_csum,
```

Comment:
In the case of the actual issue we saw, it was the dkey_csum that needs to be recalculated, is that happening here?

Reply:
Yes if I read the code correctly because we release the previous calculation above.